lib/jobs/generate-sku.js

```diff
@@ -118,13 +118,19 @@ function generateSkuJobFactory(
                     });
                 }
 
-                return waterline.nodes.updateByIdentifier(self.nodeId, {
-                    sku: match && match.sku ? match.sku.id : null
-                })
-                .then(function() {
-                    if (match && match.sku) {
-                        return eventsProtocol.publishSkuAssigned(self.nodeId, match.sku.id);
-                    }
+                return waterline.nodes.needByIdentifier(self.nodeId)
+                .then(function(oldNode) {
+                    return waterline.nodes.updateByIdentifier(self.nodeId, {
+                        sku: match && match.sku ? match.sku.id : null
+                    })
+                    .then(function(newNode) {
+                        /* asynchronous */
```

Comment:
Could you please explain more in the code you intentionally want the promise not be chained but run in background.